internal/worker/module.go

```diff
@@ -12,39 +14,55 @@ import (
 	"go.uber.org/zap"
 )
 
-const SleepTime time.Duration = 60 * time.Second
-
 func Run(lc fx.Lifecycle, cfg *config.Config, logger *zap.SugaredLogger, locker *locking.RedisLocker, stg storage.Storage) {
 	lc.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
+			redisAddress := cfg.Redis.Host + ":" + cfg.Redis.Port
+			srv := asynq.NewServer(
+				asynq.RedisClientOpt{Addr: redisAddress},
+				asynq.Config{Concurrency: cfg.MaxConcurrentTasks},
+			)
+
+			runningProcesses := &sync.Map{}
+
+			mux := asynq.NewServeMux()
+
+			startMosaicHandler := func(ctx context.Context, t *asynq.Task) error {
+				return handleStartMosaicTask(ctx, t, cfg, logger, locker, runningProcesses, stg)
+			}
+			mux.HandleFunc(TypeStartMosaic, startMosaicHandler)
+
+			logger.Info("Worker started listening for tasks")
+
 			go func() {
-				runningProcesses := make(map[string]bool)
-
-				for {
-					logger.Info("worker is running")
-
-					tasks, err := mosaic.FetchMosaicTasks(cfg.API.URL)
-					if err != nil {
-						logger.Fatal(err)
-					}
-
-					for _, task := range tasks {
-						go func(m mosaic.Mosaic) {
-							defer func() {
-								// Once finished, unmark the task
-								delete(runningProcesses, m.Name)
-							}()
-
-							if err := GenerateMosaic(m, cfg, logger, locker, &mosaic.FFMPEGCommand{}, runningProcesses, stg); err != nil {
-								logger.Error(err)
-							}
-						}(task)
-					}
-
-					time.Sleep(SleepTime)
+				if err := srv.Run(mux); err != nil {
+					logger.Fatal("Could not run asynq server: ", err)
 				}
 			}()
+
 			return nil
 		},
 	})
 }
+
+func handleStartMosaicTask(ctx context.Context, t *asynq.Task, cfg *config.Config, logger *zap.SugaredLogger, locker locking.Locker, rp *sync.Map, stg storage.Storage) error {
```

Comment:
nit: I think, you can benefit either of having a line formatted arguments
```go
func handleStartMosaicTask(
  ctx context.Context, 
  t *asynq.Task, 
  cfg *config.Config,
  logger *zap.SugaredLogger,
  locker locking.Locker,
  rp *sync.Map,
  stg storage.Storage,
) error {}
```
Or introduce a `MosaicTaskParams`.

Reply:
`MosaicTaskParams` seems a nice option!